### ISSUE TITLE
Errors using numba 0.57.0rc1

### ISSUE THREAD

**Issue by ianthomas23** [holoviz/datashader#1201]
Running CI using `numba 0.57.0rc1` gives the following errors:

```
FAILED datashader/tests/test_raster.py::test_raster_integer_nan_value - numba.core.errors.NumbaTypeError: Unsupported array type: numpy.ma.MaskedArray.
FAILED datashader/tests/test_raster.py::test_raster_integer_nan_value_padding - numba.core.errors.NumbaTypeError: Unsupported array type: numpy.ma.MaskedArray.
FAILED datashader/tests/test_raster.py::test_raster_single_pixel_range - numba.core.errors.NumbaTypeError: Unsupported array type: numpy.ma.MaskedArray.
FAILED datashader/tests/test_raster.py::test_raster_single_pixel_range_with_padding - numba.core.errors.NumbaTypeError: Unsupported array type: numpy.ma.MaskedArray.
================================= 4 failed, 897 passed, 66 skipped, 5734 warnings in 266.77s (0:04:26) =================================
```

**Comment by ianthomas23** [comment]
Thanks to @Hoxbro for first identifying this.

**Comment by ianthomas23** [comment]
Fixed by #1205.

**Comment by ianthomas23** [comment]
Closing this as numba 0.57 and python 3.11 support in now in the `main` branch. I'll cut a `dev` release later today with the expectation of a full release soon (next week?)
